python/ray/data/_internal/block_list.py

```diff
@@ -15,19 +15,27 @@ class BlockList:
     change after execution due to block splitting.
     """
 
-    def __init__(self, blocks: List[ObjectRef[Block]], metadata: List[BlockMetadata]):
+    def __init__(
+        self,
+        blocks: List[ObjectRef[Block]],
+        metadata: List[BlockMetadata],
+        owned_by_consumer: bool = False,
```

Comment:
```suggestion
        *, owned_by_consumer: bool,
```

Make ownership fields non-optional and keyword-only please, here and elsewhere.